building/building-whonix-template.md

```diff
@@ -0,0 +1,84 @@
+---
+layout: doc
+title: Building Whonix Templates
+permalink: /doc/building-whonix-template/
+redirect_from:
+- /en/doc/building-whonix-template/
+---
+
+## Building Whonix Templates
+
+The Whonix templates are easily downloaded and installed by following the [procedure here](/doc/whonix/install/).
+However, they are integrated into `qubes-builder` so they are straight-forward to build yourself if you prefer.
+
+Many other Qubes templates can also be built by following this procedure.
+Simply choose the appropriate builder(s) and template(s) you wish to build in the `./setup` procedure below.
+Always include the `mgmt-salt` builder.
+
+First, set up the [Build Environment](/doc/qubes-r3-building/#build-environment) (follow the build environment section only).
+
+Next, configure the builder:
+
+~~~
+cd ~/qubes-builder
+./setup
+# Select Yes to add Qubes Master Signing Key
+# Select Yes to add Qubes OS Signing Key
+# Select 3.2 or 4.0 for version
+# Stable
+# Yes (we want to build only templates)
+# Select builder-fedora, builder-debian, template-whonix, mgmt-salt (setup won't let you continue if you don't include builder-fedora, but we don't actually use it)
+# Choose Yes to add adrelanos's third party key
+# Yes (to download sources)
+# Select whonix-gateway, whonix-workstation (for the currently shipping templates)
+~~~
+
+Continue the build process with:
+
+~~~
+make install-deps
+make get-sources
+~~~
+
+You will often need to edit/update `qubes-src/template-whonix/builder.conf` at this stage to specify the currently shipping Tor Browser version.
+Open it in your favorite editor, then look for "Extra Whonix Build Options" and add/edit the `WHONIX_TBB_VERSION` variable to specify the current version.
```

Comment:
It is also possible to set `WHONIX_TBB_VERSION` from main `builder.conf` file (or `override.conf`, if you use one).
Not sure if worth mentioning, but generally I consider more convenient to modify just one config file, not different parts in different places.

Reply:
Good point; I'm planning on referencing back to this note from the main ISO build doc once this one is "live" so will update it then too.